at_client/lib/src/stream/stream_notification_handler.dart

```diff
@@ -39,12 +41,15 @@ class StreamNotificationHandler {
         return;
       }
       bytesReceived += onData.length;
-      var decryptedBytes = await encryptionService.decryptStream(
-          onData, streamNotification.senderAtSign);
-      f.writeAsBytesSync(decryptedBytes, mode: FileMode.append);
+      temp_file.writeAsBytesSync(onData, mode: FileMode.append);
       logger.finer('bytesReceived:${bytesReceived}');
       streamReceiveCallBack(bytesReceived);
       if (bytesReceived == streamNotification.fileLength) {
+        var encryptedData = temp_file.readAsBytesSync();
+        var decryptedBytes = await encryptionService.decryptStream(
```

Comment:
please add logger for time taken for decrypt..

Reply:
Done.

Reply:
you can create start_time and end_time. and (log end_time - start_time). Otherwise we have to manually calculate from log file

Reply:
Ok